code/modules/mob/living/simple_animal/hostile/mining/abandoned_minebot.dm

```diff
@@ -0,0 +1,88 @@
+/mob/living/simple_animal/hostile/asteroid/abandoned_minebot
+	name = "abandoned minebot"
+	desc = "The instructions printed on the side are faded, and the only thing that remains is mechanical bloodlust."
+	icon = 'icons/obj/aibots.dmi'
+	icon_state = "mining_drone_offense"
+	icon_living = "mining_drone_offense"
+	icon_aggro = "mining_drone_alert"
+	mob_biotypes = MOB_ROBOTIC
+	vision_range = 6
+	aggro_vision_range = 7
+	status_flags = CANPUSH
+	mouse_opacity = MOUSE_OPACITY_ICON
+	weather_immunities = list("ash")
+	move_to_delay = 10
+	maxHealth = 75
+	health = 75
+	melee_damage_lower = 15
+	melee_damage_upper = 15
+	obj_damage = 10
+	environment_smash = ENVIRONMENT_SMASH_STRUCTURES
+	check_friendly_fire = TRUE
+	stop_automated_movement_when_pulled = TRUE
+	attacktext = "drills into"
+	attack_sound = 'sound/weapons/circsawhit.ogg'
+	projectiletype = /obj/item/projectile/kinetic
+	projectilesound = 'sound/weapons/kenetic_accel.ogg'
+	ranged_cooldown_time = 30
+	ranged_message = "fires"
+	speak_emote = list("states")
+	del_on_death = TRUE
+	light_range = 5
+	var/sight_flags = SEE_TURFS
+	ranged = TRUE
+	retreat_distance = 2
+	minimum_distance = 1
+	deathmessage = "blows apart!"
+	loot = list(/obj/effect/gibspawner/robot, /obj/item/pickaxe/drill)
+
+/mob/living/simple_animal/hostile/asteroid/abandoned_minebot/drop_loot()
+	loot += pick(
+		/obj/item/borg/upgrade/modkit/chassis_mod,
+		/obj/item/borg/upgrade/modkit/tracer/adjustable,
+		/obj/item/borg/upgrade/modkit/cooldown,
+		/obj/item/borg/upgrade/modkit/damage,
+		/obj/item/borg/upgrade/modkit/range)
+	var/ore = pick(/obj/item/stack/ore/iron, /obj/item/stack/ore/plasma, /obj/item/stack/ore/glass/basalt)
+	var/obj/item/stack/stack = new ore(loc, rand(5, 15))
+	. = ..()
+
+/mob/living/simple_animal/hostile/asteroid/abandoned_minebot/GiveTarget(new_target)
+	if(!..())
+		return
+	if(isliving(target) && !target.Adjacent(targets_from) && ranged_cooldown <= world.time)
+		OpenFire(target)
+
+/mob/living/simple_animal/hostile/asteroid/abandoned_minebot/adjustHealth(damage, updating_health)
+	if(prob(20))
+		do_sparks(3, 1, src)
+	. = ..()
+
+/mob/living/simple_animal/hostile/asteroid/abandoned_minebot/Move(atom/newloc)
+	if(newloc && (islava(newloc) || ischasm(newloc))) //minebots aren't lava-resistant. Would be a shame if they just died.
```

Comment:
Can you check that plasma lava counts under islava()

Reply:
appears so!